internal/html/scrape/schema/json-ld/recipe_test.go

```diff
@@ -36,7 +37,22 @@ func TestRecipeProcessor_GetRecipeNode(t *testing.T) {
 			data, err := rp.GetRecipeNode(doc)
 			require.Nil(err)
 
-			require.Field("type").Equal("Recipe", data["type"])
+			recipeType := "Recipe"
+
+			if t, ok := data["type"].(string); ok {
+				require.Field("type").Equal(t, recipeType)
+			} else if t, ok := data["type"].([]interface{}); ok {
+				for _, v := range t {
+					if v == recipeType {
+						require.Equal(v, recipeType)
+						break
+					}
+					require.NotNil(nil)
+				}
+			} else {
+				require.NotNil(nil)
+			}
```

Comment:
Why use `require.NotNil(nil)`instead of `t.Fatal()`?

Also, we could just check the first element of t:



```go
			if t, ok := data["type"].(string); ok {
 				require.Field("type").Equal(t, recipeType)
 			} else if t, ok := data["type"].([]interface{}); ok {
                require.Field("type[0]").Equal(t[0], recipeType)
 			} else {
                t.Fatal("type not in expected format")
 			}
```